src/components/CreateNewRocket.svelte

```diff
@@ -46,27 +100,66 @@
 		userHasProfile = userHasProfile;
 	}
 
+	let name: string = '';
+
+	$: nameInvalid = true;
+	$: nameError = '';
+	$: needBuyName = true;
+	$: canBuyName = true;
+	$: buyNameError = '';
+	$: cannotPublish = true;
+
 	const rocketNameValidator = /^\w{4,20}$/;
-	const nameIsUnique = (name: string, rocketEvents: NDKEvent[]) => {
-		return !rocketEvents.some((event) => event.tags[0][1] === name);
+
+	const nameIsUnique = (name: string, rockets: Rocket[]) => {
+		return !rockets.some((r) => r.Name() === name);
 	};
 
-	$: if (name) {
-		if (!rocketNameValidator.test(name)) {
+	$: {
+		if (!name) {
+			nameInvalid = true;
+			nameError = '';
+		} else if (!rocketNameValidator.test(name)) {
 			nameInvalid = true;
 			nameError = 'Rocket names MUST be 4-20 alphanumeric characters';
-		} else if (!$rocketsStore) {
-			// nameInvalid = true;
-			// nameError = 'Loading Nostr';
-		} else if (!nameIsUnique(name, $rocketsStore)) {
-			// nameInvalid = true;
-			// nameError = 'Rocket names MUST be unique';
+		} else if ($rockets && !nameIsUnique(name, $rockets)) {
```

Comment:
We only need to check unique name based on the valid name purchases, duplicated names are OK as long as no one has paid for that name

Reply:
It can be understood that the testnet does not need to be unique, while the mainnet needs to be unique, right?

Reply:
I will make the necessary modifications in line with the new method of separating the mainnet and testnet mentioned below.